### DISCUSSION TITLE
Debate: Translated system prompts/ experts

### DISCUSSION THREAD

**Discussion by privacy-advo** [Kochava-Studios/witsy#239] [General]
Almost all (Western) AI models, excluding niche or fine-tuned models, are heavily biased towards English content due to their training data. This becomes particularly clear when working with resasoning models. They translate every single request into English - back and forth.

Does it make sense to have localised expert prompts (or even system prompts) in Witsy? Or does it even have a negative impact on output because of the additional steps? Should Witsy label experts localised, but keep the prompts in English?

imho: I just stick to English all the time. 

@nbonamy @jahelo @MyButtermilk

**Comment by jahelo** [comment]
I like your reasoning. In the end we probably would need to  conduct some testing. But it will depend on the model used and as you say, most current mainstream Modells will be best in English. IMHO. So yeah, sticking to English might make the most sense. And the user is still able to change the prompt if they, for whatever reason want to do that.

**Comment by MyButtermilk** [comment]
Each (big, good) model can reason in one language and still answer in another. This means the user can just add to each prompt : Answer in German.

There is no need to change anything.

I would rather add more or better prompts, like those from Schumer https://shumerprompt.com/ as they are not widely known but give high quality outputs.

**Comment by nbonamy** [comment]
Models do not translate back and forth to English. They do token prediction (that's what they do) without knowing the language they use. Of course statistically if you prompt in English, English related tokens (this does not even make real sense but anyway) have higher probabilities to be the next than German related ones do. The main difference is that their tokenizer are optimized towards English language and so they split English content more effectively.

The main challenge is this one: if a French user selects a French text and uses an expert whose prompt is in English, you cannot be certain that the LLM will respond in French. And small models will have a higher tendency to respond in English. You can understand the statistical bias that the English prompt will introduce in determining the next token.

When I implemented i18n, I had that debate with myself 😀 I finally decided to translate the expert and commands prompt because my ambition is that Witsy is used by anyone included non English speakers. And if you use a tool configured in French that responds in English then it is not a good experience.

**Comment by privacy-advo** [comment]
I made a different experience with reasoning models. Quickly loaded Phi4-mini-reasoning on a nearby laptop. First prompt:

![Phi4Reasoning](https://github.com/user-attachments/assets/6d7ab1d1-a9e7-4738-a73b-f3f8535facc5)

First it translates the prompt and later it uses an isolated German phrase in quotes within the reasoning.
Had the same experience multiple times with Qwen3 32b. (I don't have my desktop at hand for samples.)
